modules/apps/frontend-token/frontend-token-definition-impl/src/main/java/com/liferay/frontend/token/definition/internal/FrontendTokenDefinitionManager.java

```diff
@@ -0,0 +1,189 @@
+		);
+	}
+
+	public Map<String, FrontendTokenDefinitionImpl>
+		getThemeIdFrontendTokenDefinitionImpls() {
+
+		return _themeIdFrontendTokenDefinitionImpls;
+	}
+
+	public DCLSingleton<Map<String, FrontendTokenDefinitionImpl>>
+		getThemeIdFrontendTokenDefinitionImplsDCLSingleton() {
+
+		return _themeIdFrontendTokenDefinitionImplsDCLSingleton;
+	}
+
+	public void removeFrontendTokenDefinition(Long companyId, String key) {
+		if (Objects.nonNull(companyId)) {
+			_getCompanyFrontendTokenDefinitions(
+				companyId
+			).remove(
+				key
+			);
+		}
+		else {
+			_themeIdFrontendTokenDefinitionImpls.remove(key);
+		}
+	}
+
+	public void removeFrontendTokenDefinition(String key) {
+		removeFrontendTokenDefinition(null, key);
+	}
+
+	private void _addFrontendTokenDefinitionFromClientExtension(
+		Long companyId, String externalReferenceCode,
+		String frontendTokenDefinitionAsJsonString,
+		ResourceBundleLoader resourceBundleLoader) {
+
+		try {
+			Objects.requireNonNull(frontendTokenDefinitionAsJsonString);
+			Objects.requireNonNull(companyId);
+			Objects.requireNonNull(externalReferenceCode);
+
+			FrontendTokenDefinitionImpl frontendTokenDefinitionImpl =
+				_newFrontendTokenDefinitionImpl(
+					frontendTokenDefinitionAsJsonString, resourceBundleLoader);
+
+			if (frontendTokenDefinitionImpl != null) {
+				_getCompanyFrontendTokenDefinitions(
+					companyId
+				).put(
+					externalReferenceCode, frontendTokenDefinitionImpl
+				);
+			}
+		}
+		catch (JSONException jsonException) {
+			throw new RuntimeException(jsonException);
+		}
+	}
+
+	private FrontendTokenDefinitionImpl
+		_addFrontendTokenDefinitionFromWorkspace(
+			String frontendTokenDefinitionAsJsonString,
+			ResourceBundleLoader resourceBundleLoader, String themeId) {
+
+		try {
+			Objects.requireNonNull(themeId);
+
+			FrontendTokenDefinitionImpl frontendTokenDefinitionImpl =
+				_newFrontendTokenDefinitionImpl(
+					frontendTokenDefinitionAsJsonString, resourceBundleLoader,
+					themeId);
+
+			_themeIdFrontendTokenDefinitionImpls.put(
+				themeId, frontendTokenDefinitionImpl);
+
+			return frontendTokenDefinitionImpl;
+		}
+		catch (JSONException jsonException) {
+			throw new RuntimeException(jsonException);
+		}
+	}
+
+	private Map<String, FrontendTokenDefinition>
+		_getCompanyFrontendTokenDefinitions(long companyId) {
+
+		_allCompaniesFrontendTokenDefinitions.putIfAbsent(
+			companyId, new ConcurrentHashMap<>());
+
+		return _allCompaniesFrontendTokenDefinitions.get(companyId);
+	}
+
+	private FrontendTokenDefinitionImpl _newFrontendTokenDefinitionImpl(
+			String frontendTokenDefinitionAsJSON,
+			ResourceBundleLoader resourceBundleLoader)
+		throws JSONException {
+
+		return _newFrontendTokenDefinitionImpl(
+			frontendTokenDefinitionAsJSON, resourceBundleLoader, null);
+	}
+
+	private FrontendTokenDefinitionImpl _newFrontendTokenDefinitionImpl(
+			String frontendTokenDefinitionAsJSON,
+			ResourceBundleLoader resourceBundleLoader, String themeId)
```

Comment:
I believe its specific for **themeId**, since there is a _getThemeId_() in  **FrontendTokenDefinitionImpl**. We could add a getExternalReferenceCode() as well, but it would not have a usage for now. We can add in a posterior moment, in case someone needs it